### PR TITLE
tools: add `source` field (Loc)

### PR DESCRIPTION
**Pull request by aspeddro**
Emit location of type/value/module

```jsonc
{filepath: "src/A.resi", line: 2, col: 2}
```

In general, you can run the command from anywhere. The `filepath` property is relative to the package root.

### CODE REVIEW

**Comment by zth** [comment]
Missing ReScript bindings here, right? Should add them to the ReScript files.

**Comment by aspeddro** [comment]
ReScript files updated!

File: tools/src/tools.ml [hunk 1]
```diff
@@ -257,7 +280,40 @@ let typeDetail typ ~env ~full =
...
+    match dirname dirPath = realPath with
+    | true -> basename dirPath :: acc |> String.concat "/"
+    | false -> (
+      match dirPath with
+      | "/" -> failwith "Failed to find relative path of package"
+      | "." -> path
+      | _ -> loop (dirname dirPath) (basename dirPath :: acc))
+  in
+
+  loop path []
```
Comment (zth) [inline tools/src/tools.ml:304]: No built in fn that does this I take it...?

Comment (aspeddro) [reply tools/src/tools.ml]: thank you. Found in `Analysis.Files.realpath`

5396f79b

Comment (zth) [reply tools/src/tools.ml]: Great! Now we just need to figure out the tests for Windows. Paths being different is obviously expected.

**Review by zth — APPROVED** [review]

**Pull request merged**